modules/core/04-channel/keeper/upgrade.go

```diff
@@ -331,6 +331,80 @@ func (k Keeper) WriteUpgradeAckChannel(ctx sdk.Context, portID, channelID string
 	emitChannelUpgradeAckEvent(ctx, portID, channelID, channel, upgrade)
 }
 
+// ChanUpgradeConfirm is called on the chain which is on FLUSHING after chanUpgradeAck is called on the counterparty.
+// This will inform the TRY chain of the timeout set on ACK by the counterparty. If the timeout has already exceeded, we will write an error receipt and restore.
+func (k Keeper) ChanUpgradeConfirm(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	counterpartyChannelState types.State,
+	counterpartyUpgrade types.Upgrade,
+	proofChannel,
```

Comment:
not sure if it's worth making the naming explicit here: `proofCounterpartyChannel`, `proofCounterpartyUpgrade` as is used below

Reply:
personally I find it easier to read through when its the shorter form of the name, you're never really verifying a proof of yourself, you're always verifying some counterparty state so I think its implicit 🤷 

if everyone prefers having the `counterparty` prefix tho, we can change them all in a follow up